cypress/pageobject/Users/UserCreation.ts

```diff
@@ -1,45 +1,122 @@
+  typeFirstName(firstName: string) {
+    cy.get("#firstName").click().type(firstName);
   }
-  clearIntoElementById(elementId: string) {
-    cy.get("#" + elementId)
-      .click()
-      .clear();
+  typeLastName(lastName: string) {
+    cy.get("#lastName").click().type(lastName);
+  }
+  typeEmail(email: string) {
+    cy.get("#email").click().type(email);
+  }
+
+  typePassword(password: string) {
+    cy.get("#password").click().type(password);
+  }
+  typePhoneNumber(phone_number: string) {
+    cy.get("#phoneNumber").click().type(phone_number);
+  }
+  typeConfirmPassword(password: string) {
+    cy.get("#c_password").click().type(password);
+  }
+  typeQualification(qualification: string) {
+    cy.get("#qualification").click().type(qualification);
+  }
+  typeDoctorExperience(experience: string) {
+    cy.get("#doctor_experience_commenced_on").click().type(experience);
+  }
+
+  typeDoctorMedicalCouncilRegNo(regNo: string) {
+    cy.get("#doctor_medical_council_registration").click().type(regNo);
+  }
+
+  typeWeeklyWorkingHours(workingHrs: string) {
+    cy.get("#weekly_working_hours").click().type(workingHrs);
+  }
+
+  typeAltPhoneNumber(emergencyPhoneNumber: string) {
+    cy.get("#altPhoneNumber").click().type(emergencyPhoneNumber);
   }
 
   typeIntoInputByName(inputName: string, value: string) {
     cy.get("input[name='" + inputName + "']")
       .click()
       .type(value);
   }
+  typeDateOfBirth(dob: string) {
+    cy.clickAndTypeDate("#date_of_birth", dob);
+  }
+  typeNewUserFirstName(firstName: string) {
+    cy.get("#first_name").click().type(firstName);
+  }
+  typeNewUserLastName(lastName: string) {
+    cy.get("#last_name").click().type(lastName);
+  }
+  typeNewUserPhoneNumber(phoneNumber: string) {
+    cy.get("#phone_number").click().type(phoneNumber);
+  }
 
-  selectOptionContainingText(text: string) {
-    cy.get("[role='option']").contains(text).click();
+  clearFirstName() {
+    cy.get("#firstName").click().clear();
+  }
+  clearLastName() {
+    cy.get("#lastName").click().clear();
+  }
+  clearPhoneNumber() {
+    cy.get("#phoneNumber").click().clear();
+  }
+  clearAltPhoneNumber() {
+    cy.get("#altPhoneNumber").click().clear();
+  }
+  clearWeeklyWorkingHours() {
+    cy.get("#weekly_working_hours").click().clear();
+  }
+  clearEmail() {
+    cy.get("#email").click().clear();
   }
 
-  verifyNotification(message: string) {
-    cy.verifyNotification(message);
+  selectUserType(role: string) {
+    cy.clickAndSelectOption("#user_type", role);
+  }
+  selectHomeFacility(name: string) {
+    cy.clickAndSelectOption("#home_facility", name);
   }
 
+  selectGender(gender: string) {
+    cy.clickAndSelectOption("#gender", gender);
+  }
+  selectState(state: string) {
+    cy.clickAndSelectOption("#state", state);
+  }
+  selectDistrict(district: string) {
+    cy.clickAndSelectOption("#district", district);
+  }
   selectFacility(name: string) {
     this.typeIntoInputByName("facilities", name);
```

Comment:
- use the reusable function available in commands.ts file, to select facilities from dropdown